R/functions-Spectrum1.R

```diff
@@ -93,12 +93,15 @@ Spectra1_mz_sorted <- function(peaksCount = NULL, rt = numeric(),
                                intensity = numeric(), fromFile = integer(),
                                centroided = NA, smoothed = NA,
                                polarity = NA_integer_, nvalues = integer()) {
-    if (length(mz) == 0 | length(intensity) == 0 | length(nvalues) == 0) {
-        stop("Arguments 'mz', 'intensity' and 'nvalues' are required!")
-    } else {
-        if (length(mz) != length(intensity))
-            stop("Lengths of 'mz' and 'intensity' do not match!")
-    }
+    ## Fix issue #215: remove check to allow empty spectra
+    ## if (length(mz) == 0 | length(intensity) == 0 | length(nvalues) == 0) {
+    ##     stop("Arguments 'mz', 'intensity' and 'nvalues' are required!")
+    ## }
+    if (sum(nvalues) != length(mz))
```

Comment:
`nvalues` is an `integer` of length equal to the number of spectra, each value representing the number of peaks it contains (and hence how many `mz` and `intensity` values it contains). This is used to *split* the `mz` and `intensity` vectors (that contain the mz and intensity values of all spectra) per spectrum. 
The test is thus OK.